### PR TITLE
Module identification

### PR DESCRIPTION
**Pull request by LRstudentHU**

### CODE REVIEW

**Review by itzandroidtab — APPROVED** [review]

File: code/headers/base_comm.hpp [hunk 1]
```diff
@@ -244,7 +250,16 @@ namespace r2d2 {
          * @param frame
          */
         void accept_frame(const frame_s &frame) {
-            rx_buffer.push(frame);
+            // Identity frames are handled here explicitly,
+            // as we want to have access to higher level sending
+            // functionality but don't want the identity request
+            // frame to get into the rx_buffer of the specific module.
+            if (frame.request && frame.type == frame_type::IDENTITY) {
```
Comment (itzandroidtab) [inline code/headers/base_comm.hpp:248]: Is it not better to change the order around?

First check for a `frame.type == frame_type::IDENTITY` as that occurs less?

File: test/main.cpp [hunk 2]
```diff
@@ -56,3 +54,25 @@ TEST_CASE("Accept all packets works", "[bus]") {
...
 }
+
+TEST_CASE("Identity requests get answered implicitly", "[bus]") {
+    mock_comm_c basic_module;
+
+    basic_module.configure(module::TEST, {
+        frame_type::ALL
+    });
+
+    auto req = basic_module.create_frame<frame_type::IDENTITY>();
```
Comment (itzandroidtab) [inline test/main.cpp:65]: Maybe we should add some extra documentation as the `create_frame<>()` is only once in the documentation

Comment (LRstudentHU) [reply test/main.cpp:65]: Sure, but that is not part of this PR.

File: code/headers/base_comm.hpp (hunk 1 above)
Comment (LRstudentHU) [reply code/headers/base_comm.hpp:248]: Technically yes, but it honestly doesn't matter at all for performance.

**Pull request merged**